src/bootstrap/src/core/builder/tests.rs

```diff
@@ -5,6 +5,10 @@ use crate::Flags;
 use crate::core::build_steps::doc::DocumentationFormat;
 use crate::core::config::Config;
 
+static TEST_TRIPLE_1: &str = "i686-unknown-haiku";
+static TEST_TRIPLE_2: &str = "i686-unknown-hurd-gnu";
+static TEST_TRIPLE_3: &str = "i686-unknown-netbsd";
+
```

Comment:
`cc` now requires real targets, so these are used instead of `A-A`. However, if I use tier 1/2 targets, I get failures like:
```diff
thread 'core::builder::tests::defaults::build_cross_compile' panicked at src/core/builder/tests.rs:301:9:
assertion failed: `(left == right)`

Diff < left / right > :
 [
     Std {
-        target: aarch64-unknown-linux-gnu,
+        target: x86_64-unknown-linux-gnu,
         compiler: Compiler {
             stage: 0,
             host: x86_64-unknown-linux-gnu,
         },
         crates: [],
         force_recompile: false,
         extra_rust_args: [],
         is_for_mir_opt_tests: false,
     },
```

Tier 3 works fine, but this seems like some weird behaviour

Reply:
cc https://rust-lang.zulipchat.com/#narrow/channel/326414-t-infra.2Fbootstrap/topic/Weird.20error.20when.20using.20T1.2FT2.20targets.20in.20tests

Reply:
I cannot reproduce the failure locally. What OS/HW arch do you have?

Reply:
Hmm, okay. I have the same target, but `x test bootstrap` passes for me locally.

Reply:
[demo.txt](https://github.com/user-attachments/files/17848707/demo.txt)
Hmm try applying this then running `x t src/bootstrap -- build_cross_compile `

Reply:
Thanks, now I can reproduce it, I misunderstood the instructions before.

So, there are some warnings about "family detection failures", which is caused by `cc`, which uses a "script" (C program) called `detect_compiler_family.c` to detect if the given target can be used on a given platform. The problem is that you don't have an aarch64 gcc cross-compiler available, so the detection fails and the target regresses back to x86. But I'm not sure if that is an actual problem for bootstrap.

Regarding the test, it looks like the steps are being produced in reverse order, for some reason. If you switch the order like this:
```rust
        assert_eq!(
            first(cache.all::<compile::Std>()).as_slice(), &[
                std!(TEST_TRIPLE_1 => TEST_TRIPLE_2, stage = 0),
                std!(TEST_TRIPLE_1 => TEST_TRIPLE_2, stage = 1),
                std!(TEST_TRIPLE_1 => TEST_TRIPLE_1, stage = 0),
                std!(TEST_TRIPLE_1 => TEST_TRIPLE_1, stage = 1),
            ]
        );
```
then it passes.

Reply:
> But I'm not sure if that is an actual problem for bootstrap.

Hmm, probably not - but if we now have to use real target triples, this test could start confusingly failing for everyone who doesn't have the right tools installed. I think sticking with these targets is probably the best course of action then

Reply:
Ok. So leave the PR as is?